src/CPUSolver.cpp

```diff
@@ -135,6 +135,9 @@ void CPUSolver::initializeFluxArrays() {
     size = _num_FSRs * _num_groups;
     _scalar_flux = new FP_PRECISION[size];
     _old_scalar_flux = new FP_PRECISION[size];
+    
+    /* Set the old scalar flux to zero */
+    memset(_old_scalar_flux, 0.0, sizeof(FP_PRECISION) * size);
```

Comment:
I agree that this initialization is needed. But I think it would be better done by updating the ``Solver::flattenFSRFluxes()`` routine to set *both* ``_scalar_flux`` and ``_old_scalar_flux`` to 0.